accuracy_metric.py

```diff
@@ -0,0 +1,161 @@
+"""
+Automatic accuracy metric for MT-GenEval paper.
+We use the correct (gender-matched) references and incorrect (counterfactual/swapped) references to evaluate accuracy.
+We first check the references for unique/gender-specific terms, then check for overlap with the incorrect gender
+and the hypothesis. If there is overlap, we mark the segment as incorrect.
+"""
+import string
+import argparse
+
+
+STRIP_PUNCT = str.maketrans(string.punctuation, ' '*len(string.punctuation))
+
+
+def clean_line(line):
+    """
+    For an input line, lowercase it, strip extra spaces, and replace punctuation with spaces.
+
+    :param line: Line to clean.
+    :return: Cleaned version of the line
+    """
+    return line.lower().translate(STRIP_PUNCT).strip()
+
+
+def get_words(line):
+    """
+    Helper function to get the set of words in a line.
+
+    :param line: Line from which to get the words.
+    :return: Set of words in the line.
+    """
+    return set(line.strip().split())
+
+
+def get_trg_correct_incorrect(cor_words, inc_words, trg_words):
+    """
+    Compute overlap between references and translation
+    """
+    cor_unique = cor_words - inc_words
+    inc_unique = inc_words - cor_words
+    # now check the words in the target sentence for overlap with incorrect unique words
+    trg_correct = trg_words & cor_unique 
+    trg_incorrect = trg_words & inc_unique
+    return trg_correct, trg_incorrect 
+
+
+def gender_decision(trg_line, cor_line, inc_line):
+    """
+    Check if gender of a sentence is correct based on corresponding correct and incorrect references.
+    Algorithm: We make decision based on whether hyp overlaps with original ref and counterfactual ref
+
+    :param trg_line: Sentence from translation output for which to check gender.
+    :param cor_line: Feminine reference.
+    :param inc_line: Masculine reference.
+    :return: a list of decision, overlap(hyp, original ref), overlap(hyp, counterfactual ref)
+    """
+    # start by getting unique words in each of the references
+    cor_words, inc_words = get_words(cor_line), get_words(inc_line)
+    trg_words = get_words(trg_line)
+    trg_correct, trg_incorrect = get_trg_correct_incorrect(cor_words, inc_words, trg_words)
```

Comment:
Since `get_trg_correct_incorrect` always is operated on the words, maybe call `get_words(cor_line)`, `get_words(inc_line)`, and `get_words(trg_line)` within that function.